### ISSUE TITLE
Enable multi-architecture support, specifically for s390x, in CNAO and the associated CNIs.

### ISSUE THREAD

**Issue by ashokpariya0** [kubevirt/cluster-network-addons-operator#1916]
**Is your feature request related to a problem? Please describe**:
Provide multi-platform support for CNAO and all CNIs supported by CNAO.

**Describe the solution you'd like**:
We would like to see multi-architecture support for CNAO and all of its supported CNIs, including:

multus
multusDynamicNetworks
linuxBridge
kubeMacPool
ovs
macvtap
kubeSecondaryDNS
kubevirtIpamController

**Describe alternatives you've considered**:

**Comment by ormergi** [comment]
@ashokpariya0 could you please elaborate on the motivation for supporting s390x arch? 
Is there some tracking issue or enhancement proposal for supporting s390x?

**Comment by ashokpariya0** [comment]
We are enabling s390x (IBM Z system) support in all(most) kubevirt ecosystem components. Already [kubevirt](https://github.com/kubevirt/kubevirt/pull/10490) (virt-operator), supports s390x and now we are working on additional components like cnao, cdi(almost done), ssp, etc.

**Comment by oshoval** [comment]
Checked again, wrt auto bumper, i think in this specific case we dont need to update bumper scripts as this parameter is part of CNAO and not part of the components.
So your PR is fine.

What about all the other images, some of them arent multiarch right ?
We won't able to maintain it right now, but will prioritize it as a task for all the images once we can.
If you plan to contribute and able to give support if it breaks we might able to take your PR, thanks.

Also please consider that kube-rbac-proxy need to be changed on https://github.com/k8snetworkplumbingwg/kubemacpool so if KMP is installed standalone or via CNAO it will be the same

WRT the proposed image, it seems safe to use it as https://github.com/brancz/kube-rbac-proxy is the one
https://github.com/openshift/kube-rbac-proxy is forked from

@phoracek agree ? anything i missed ?
Thanks

**Comment by ashokpariya0** [comment]
> Checked again, wrt auto bumper, i think in this specific case we dont need to update bumper scripts as this parameter is part of CNAO and not part of the components. So your PR is fine.
> 
>   @oshoval Thank you for your feedback. 
> What about all the other images, some of them arent multiarch right ? We won't able to maintain it right now, but will prioritize it as a task for all the images once we can. If you plan to contribute and able to give support if it breaks we might able to take your PR, thanks. Maybe we first need to map all the required changes over all repos and just then start by fixing one by one.
> 
> Yes, you are right. We have noticed that most of the CNI images do not support multiple architectures, and we are in the process of identifying the necessary changes. Our initial focus is to get the CNAO operator running on the s390x architecture(for this we need to change kube-rbac-proxy image), and then we will address the other CNI images one by one, as you suggested. We are committed to actively contributing to this effort.
> Also please consider that kube-rbac-proxy need to be changed on https://github.com/k8snetworkplumbingwg/kubemacpool so if KMP is installed standalone or via CNAO it will be the same
> 
> Okay, Noted.
> WRT the proposed image, it seems safe to use it as https://github.com/brancz/kube-rbac-proxy is the one https://github.com/openshift/kube-rbac-proxy is forked from
> 
> Yes, That is correct.
> Wdyt about fixing it / opening an issue on `https://github.com/openshift/kube-rbac-proxy` please ?
> 
>  Sure, I will create the issue and follow up on it.

**Comment by oshoval** [comment]
> Yes, you are right. We have noticed that most of the CNI images do not support multiple architectures, and we are in the process of identifying the necessary changes. Our initial focus is to get the CNAO operator running on the s390x architecture(for this we need to change kube-rbac-proxy image), and then we will address the other CNI images one by one, as you suggested. We are committed to actively contributing to this effort.

Thank you, just to make sure, i meant other components that CNAO deploys, do we talk on the same?
```
  kubeMacPool: {}
  kubeSecondaryDNS: {}
  kubevirtIpamController: {}
  linuxBridge: {}
  macvtap: {}
  multus: {}
  multusDynamicNetworks: {}
  ovs: {}
```

**Comment by ashokpariya0** [comment]
> Thank you
> 
> > Yes, you are right. We have noticed that most of the CNI images do not support multiple architectures, and we are in the process of identifying the necessary changes. Our initial focus is to get the CNAO operator running on the s390x architecture(for this we need to change kube-rbac-proxy image), and then we will address the other CNI images one by one, as you suggested. We are committed to actively contributing to this effort.
> 
> Just to make sure, i meant other components that CNAO deploys, do we talk about the same?
> 
> ```
> kubeMacPool
> kubeSecondaryDNS
> kubevirtIpamController
> linuxBridge
> macvtap
> multus
> multusDynamicNetworks
> ovs
> ```

Yes.
I tried all of the above CNIs, Below is my findings.
**Multus** : Able to deploy  with latest image tag.
**Multus Dynamic Networks Controller**: We don't have mutli arch supported image for this CNI plugin. We need to build one.
**linuxBridge**- Image is multi arch supported but we see runtime issue(Marker binary is unavailable for s390x) with images.
**macvtap:** Image is multi arch supported but cp binray present inside image is not s390x compatible
**ovs:** We don't mutli arch images for this CNI plugin.
**KubeSecondaryDNS** : We don't multi arch image for this CNI plugin.

**Comment by oshoval** [comment]
amazing thank you
what about kubeMacPool / kubevirtIpamController ?

**Comment by oshoval** [comment]
Please keep in mind for changes that are done, that we will also need ARM, so best if changes can be done in a robust manner (or at least strives to)

**Comment by ashokpariya0** [comment]
> Amazing thank you what about kubeMacPool / kubevirtIpamController if possible as well please ?

Regarding **kubemacpool:**- Image is multi arch supported However I see runtime issue with image as manager binary present inside kubemacpool-cert-manager init container is not compatible with s390x.
**kubevirtIpamController**: looks fine.

**Comment by oshoval** [comment]
Thanks a lot, we are here whatever needed

**Comment by oshoval** [comment]
Hi @zhlhahaha 
Might interest you about ARM

**Comment by oshoval** [comment]
Might worth please to consider a gist about how to emulate s390x so whoever need to sanity check it can
https://www.qemu.org/docs/master/system/target-s390x.html

**Comment by ashokpariya0** [comment]
> Might worth please to consider a gist about how to emulate s390x so whoever need to sanity check / develop it can https://www.qemu.org/docs/master/system/target-s390x.html

Sure, Thanks!

**Comment by ashokpariya0** [comment]
@oshoval  Please check PR: https://github.com/kubevirt/cluster-network-addons-operator/pull/1917

**Comment by ashokpariya0** [comment]
@oshoval I created an issue on [GitHub](https://github.com/openshift/kube-rbac-proxy) to enable multi-architecture image support for quay.io/openshift/origin-kube-rbac-proxy. You can find the issue here: [Issue #113](https://github.com/openshift/kube-rbac-proxy/issues/113)

**Comment by ashokpariya0** [comment]
@oshoval  @RamLavi The last pushed upstream CNAO image ([link](https://quay.io/repository/kubevirt/cluster-network-addons-operator?tab=tags&tag=latest)) was 13 days ago. Since then, a couple of pull requests have been merged in the CNAO repository (specifically the last two). However, I haven't seen a new operator image published. Could you please help clarify the reason for this?

**Comment by oshoval** [comment]
Hi, we dont release on each PR, just once there is a need, if you need it we can release

**Comment by ashokpariya0** [comment]
@phoracek @oshoval @RamLavi 
Please review the PR at https://github.com/kubevirt/cluster-network-addons-operator/pull/1923 when you get a chance.

**Comment by ashokpariya0** [comment]
@oshoval @phoracek could you please direct me to the GitHub repository where the image below is being prepared?
https://quay.io/repository/kubevirt/cni-default-plugins?tab=tags&tag=latest

**Comment by oshoval** [comment]
good question
according https://github.com/kubevirt/cluster-network-addons-operator/issues/636
which point to https://github.com/kubevirt/cluster-network-addons-operator/pull/595
Petr might know please ?

**Comment by oshoval** [comment]
hack/components/bump-linux-bridge.sh
by this script i think, which runs on CNAO bump actions

**Comment by ashokpariya0** [comment]
> hack/components/bump-linux-bridge.sh by this script i think, which runs on CNAO git actions (bump release action)

Thanks, @oshoval , for the quick reply! I got my answer.

**Comment by oshoval** [comment]
you mean who build quay.io/openshift/origin-operator-registry:4.2.0 ?
i dont know by heart sorry, if i find i will update here

**Comment by oshoval** [comment]
```
$ docker inspect quay.io/openshift/origin-operator-registry | grep git
                "io.openshift.build.source-location": "https://github.com/openshift/operator-framework-olm",
                "vcs-type": "git",
                "vcs-url": "https://github.com/openshift/operator-framework-olm"
                "io.openshift.build.source-location": "https://github.com/openshift/operator-framework-olm",
                "vcs-type": "git",
                "vcs-url": "https://github.com/openshift/operator-framework-olm"
```

does this help ?

**Comment by ashokpariya0** [comment]
> ```
> $ docker inspect quay.io/openshift/origin-operator-registry | grep git
>                 "io.openshift.build.source-location": "https://github.com/openshift/operator-framework-olm",
>                 "vcs-type": "git",
>                 "vcs-url": "https://github.com/openshift/operator-framework-olm"
>                 "io.openshift.build.source-location": "https://github.com/openshift/operator-framework-olm",
>                 "vcs-type": "git",
>                 "vcs-url": "https://github.com/openshift/operator-framework-olm"
> ```
> 
> does this help ? repo fits the one you said
> 
> https://github.com/search?q=repo%3Aopenshift%2Foperator-framework-olm%20This%20is%20a%20component%20of%20OpenShift%20Operator%20Lifecycle%20Manager%20and%20is%20the%20base%20for%20operator%20catalog%20API%20containers.&type=code seems you are correct according the desc field
> 
> ```
> $ docker inspect quay.io/openshift/origin-operator-registry | grep description
>                 "io.k8s.description": "This is a component of OpenShift Operator Lifecycle Manager and is the base for operator catalog API containers.",
>                 "io.k8s.description": "This is a component of OpenShift Operator Lifecycle Manager and is the base for operator catalog API containers.",
> ```

Yes, Thanks.

**Comment by ashokpariya0** [comment]
@oshoval @phoracek @RamLavi  
I created a PR [[#77](https://github.com/kubevirt/bridge-marker/pull/77)](https://github.com/kubevirt/bridge-marker/pull/77) to enable multi-platform/architecture support for the `bridge-marker` image.

**Comment by ashokpariya0** [comment]
Added  multi-platform/architecture support for the `kubemacpool` image.
 PR [[#441](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441)]
 @oshoval

**Comment by ashokpariya0** [comment]
@oshoval 
Can we add the s390x platform alongside amd64 in the [Makefile](https://github.com/kubevirt/cluster-network-addons-operator/blob/302ad3ea1f3bce6a23c2bec86a7e58a31ee2f516/Makefile#L16), similar to how it's done in the [ipam-extensions Makefile](https://github.com/kubevirt/ipam-extensions/blob/925bee6139751fcb241a435bae8b79ce1dfce595/Makefile#L121)?
By default, it will only build for the amd64 architecture unless s390x is explicitly included. Additionally, we need to remember to export the PLATFORMS variable like so:
export PLATFORMS=linux/arm64,linux/s390x,linux/amd64
I think it would be a good idea to add s390x for now, and we can include other platforms later. I can also add a note to the developer workflow, advising how to set or unset this platform configuration

**Comment by oshoval** [comment]
> Added multi-platform/architecture support for the `kubemacpool` image. PR [[#441](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441)] @oshoval

@RamLavi fyi please

**Comment by oshoval** [comment]
Very nice thanks

**Comment by oshoval** [comment]
> @oshoval Can we add the s390x platform alongside amd64 in the [Makefile](https://github.com/kubevirt/cluster-network-addons-operator/blob/302ad3ea1f3bce6a23c2bec86a7e58a31ee2f516/Makefile#L16), similar to how it's done in the [ipam-extensions Makefile](https://github.com/kubevirt/ipam-extensions/blob/925bee6139751fcb241a435bae8b79ce1dfce595/Makefile#L121)? By default, it will only build for the amd64 architecture unless s390x is explicitly included. Additionally, we need to remember to export the PLATFORMS variable like so: export PLATFORMS=linux/arm64,linux/s390x,linux/amd64 I think it would be a good idea to add s390x for now, and we can include other platforms later. I can also add a note to the developer workflow, advising how to set or unset this platform configuration

Personally i prefer not, at least not in this stage, lets first finish the first phase of adapting the components and the release lanes and then we can consider ?
I might be wrong here, best to ask for more opinions

**Comment by ashokpariya0** [comment]
@RamLavi @phoracek @oshoval 
I have submitted a PR for multi-platform/architecture support. Please review the following:

For Linux Bridge:
Bridge-Marker: [PR #77](https://github.com/kubevirt/bridge-marker/pull/77)
Linux-Bridge: [PR #1927](https://github.com/kubevirt/cluster-network-addons-operator/pull/1927)

For KubeMacPool:
[PR #441](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441)

**Comment by ashokpariya0** [comment]
> Thanks FYI if you want to check this one please [k8snetworkplumbingwg/ovs-cni#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336)

Sure

**Comment by ashokpariya0** [comment]
> > @oshoval Can we add the s390x platform alongside amd64 in the [Makefile](https://github.com/kubevirt/cluster-network-addons-operator/blob/302ad3ea1f3bce6a23c2bec86a7e58a31ee2f516/Makefile#L16), similar to how it's done in the [ipam-extensions Makefile](https://github.com/kubevirt/ipam-extensions/blob/925bee6139751fcb241a435bae8b79ce1dfce595/Makefile#L121)? By default, it will only build for the amd64 architecture unless s390x is explicitly included. Additionally, we need to remember to export the PLATFORMS variable like so: export PLATFORMS=linux/arm64,linux/s390x,linux/amd64 I think it would be a good idea to add s390x for now, and we can include other platforms later. I can also add a note to the developer workflow, advising how to set or unset this platform configuration
> 
> Personally i prefer not, at least not in this stage, lets first finish the first phase of adapting the components and the release lanes and then we can consider ? I might be wrong here, best to ask for more opinions
> 
>

@oshoval Regarding multiplatform support, it's not ideal yet, but we can begin by adding support for AMD64 and s390x architectures. I've already completed testing for these, so once multiplatform support is integrated for CNAO-supported CNIs and the release is finalized, we’ll just need to update the image digest.

Currently, the Multus CNI already supports multiplatform, so we can directly proceed with testing that.
I think it would be a great idea to start publishing multiplatform-supported images for the CNAO operator starting with the next release. This would enable us to test upstream images on the s390x architecture as well.
What do you think ? @phoracek @RamLavi 


> btw what about auto detection ? it can win both worlds, be light and automatic
>
Yes, this is a good suggestions, and I can submit the PR for it.

**Comment by ashokpariya0** [comment]
> btw what about auto detection ? it can win both worlds, be light and automatic
>
Added in PR: https://github.com/kubevirt/cluster-network-addons-operator/pull/1928

**Comment by LakshmiRavichandran1** [comment]
@RamLavi  @phoracek @oshoval ,
thank you for your support from the community to enable s390x efforts on CNAO and the related plugins.

- I take this opportunity to increase the importance of the below s390x PRs from @ashokpariya0  and the urge to get them merged before the feature freeze on 02.Dec.2024 (coming monday).

- Additionally, is it possible to get a new release created for each of the CNIs, so that we can update their image digests in the CNAO codebase before 02.Dec?

For your kind visibility, these are the PRs:

For Multus Dynamic:
----------------------
https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller/pull/267

For Linux Bridge:
-------------------
Bridge-Marker: https://github.com/kubevirt/bridge-marker/pull/77
Linux-Bridge: https://github.com/kubevirt/cluster-network-addons-operator/pull/1927

For KubeMacPool:
-------------------
https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441

**Comment by oshoval** [comment]
Thanks for all your contribution.
Sorry, we won't able to finish it by tomorrow, it should get reviewed and prioritized,
we are doing the best we can.

**Comment by oshoval** [comment]
There are also those PRs by @jschintag
https://github.com/kubevirt/kubesecondarydns/pull/73
https://github.com/kubevirt/kubesecondarydns/pull/74
https://github.com/kubevirt/kubesecondarydns/pull/75

Thanks

**Comment by oshoval** [comment]
From my point of view we must hold until https://github.com/kubevirt/cluster-network-addons-operator/pull/1923#issuecomment-2509657197 is addressed fully

also i am going to try CNAO locally, i might have an older fedora and will need time to make sure it runs locally (or that i need to upgrade which will take time)
I am on other tasks as well, and won't able to maintain CNAO otherwise

**Comment by oshoval** [comment]
about https://github.com/kubevirt/cluster-network-addons-operator/issues/1916#issuecomment-2498527899
I thought about it again, auto detection is fine, compiling both is not, our time is precious and build should be fast
Please consider that on the mentioned PRs that sometimes suggest dual compilation instead auto detection

**Comment by ashokpariya0** [comment]
> about [#1916 (comment)](https://github.com/kubevirt/cluster-network-addons-operator/issues/1916#issuecomment-2498527899) I thought about it again, auto detection is fine, compiling both is not, our time is precious and build should be fast Please consider that on the mentioned PRs (unless already done)
> 
> (for completeness you might want to benchmark with and without on worst case, and if it is neglectable we can theoretically consider it, but i don't think it make sense to force it)
>
I've noticed that multi-platform builds are much faster with Docker Buildx since it builds in parallel, whereas with Podman, it takes more time (although I can see ways to optimize that). Additionally, could we consider adding a GitHub Action to offload the builds to a CI/CD system for better performance in multi-platform scenarios?

**Comment by oshoval** [comment]
Hi, about git actions not atm please, won't have time to work on that, lets first do the minimum first
about compilation, we can have benchmark and act accordingly please once we can compare, imho auto detection is good enough in any case (let me know please why not if you think otherwise)

**Comment by oshoval** [comment]
All your PRs that still arent merged, so will be easier to track, if i miss anything / you would like to add new ones please let me know

Thanks a lot for your effort

- [ ] [kubevirt/kubesecondarydns#75](https://github.com/kubevirt/kubesecondarydns/pull/75)
- [ ] [kubevirt/kubesecondarydns#73](https://github.com/kubevirt/kubesecondarydns/pull/73)
- [ ] [k8snetworkplumbingwg/kubemacpool#441](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441)
- [ ] [kubevirt/cluster-network-addons-operator#1927](https://github.com/kubevirt/cluster-network-addons-operator/pull/1927)
- [ ] [kubevirt/cluster-network-addons-operator#1928](https://github.com/kubevirt/cluster-network-addons-operator/pull/1928)
- [ ] [kubevirt/bridge-marker#77](https://github.com/kubevirt/bridge-marker/pull/77)
- [ ] [k8snetworkplumbingwg/ovs-cni#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336)

**Comment by oshoval** [comment]
We miss macvtap in case you would like to address it please as well
Thanks

**Comment by ashokpariya0** [comment]
> We miss macvtap in case you would like to address it please as well Thanks
>
Sure Noted, I will work on that.

**Comment by ashokpariya0** [comment]
@oshoval Once the PR [#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336) gets merged, I can submit the changes for s390x (which will be a small update). After that, we can request a new release for the OVS CNI.

**Comment by ashokpariya0** [comment]
> > @oshoval Once the PR [#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336) gets merged, I can submit the changes for s390x (which will be a small update). After that, we can request a new release for the OVS CNI.
> 
> awesome thanks [k8snetworkplumbingwg/ovs-cni#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336) is merged
>
I submitted s390x support PR for OVS CNI: https://github.com/k8snetworkplumbingwg/ovs-cni/pull/337

**Comment by oshoval** [comment]
Thanks

Lets please also run locally docker / podman on the repos to make sure it does work for both for multi arch / cross etc
at least some of the combinations

if help needed lets sync on slack, thanks

**Comment by ashokpariya0** [comment]
> Thanks
> 
> Lets please also run locally docker / podman on the repos to make sure it does work for both for multi arch / cross etc at least some of the combinations
> 
> if help is needed lets sync on slack, thanks
>
Thanks @oshoval 
I’ve considered that, and here’s what I do:
1) I build on amd64 and then test on s390x to validate cross-compilation.
2) I also build directly on s390x and test on the same architecture to ensure everything works natively.
3) Additionally, I test with both Docker and Podman to ensure compatibility across both container runtimes.

**Comment by oshoval** [comment]
Awesome, thanks a lot

**Comment by oshoval** [comment]
Hi, I didn't manage yet to have a system end to end that compiles and push everything in every combinations, 
and seems i won't able to focus on that, so i can't maintain it myself atm.
It means that if problems related to multi arch will arise, we might need to ask your help,
will you able please to support it?
Our main effort will still be the stability of the main flow releases.
Thanks

**Comment by oshoval** [comment]
/cc @jschintag
since you helped in this effort, thanks

**Comment by ashokpariya0** [comment]
> Hi, I didn't manage yet to have a system end to end that compiles and push everything in every combinations,
> and seems i won't able to focus on that, so i can't maintain it myself atm.
> It means that if problems related to multi arch will arise, we might need to ask your help,
> will you able please to support it?
> Our main effort will still be the stability of the main flow releases.
> Thanks
>
@oshoval  We're here to assist whenever you need extra hands with multi-arch or any related concerns. Just let us know, and we'll do our best to help!

**Comment by oshoval** [comment]
Thank you very much

**Comment by oshoval** [comment]
please look on post submit of cnao under project infra or on git actions
if you dont manage i will try to help

**Comment by ashokpariya0** [comment]
> Can you please check that CNAO release jobs themselves support and will use "all" flag ?
>
Done, PR link: https://github.com/kubevirt/cluster-network-addons-operator/pull/1956

**Comment by ashokpariya0** [comment]
> All your PRs that aren't merged yet, so will be easier to track, if i miss anything / you would like to add new ones please let me know
> 
> Thanks a lot for your effort
> 
> * [x]  [kubevirt/kubesecondarydns#75](https://github.com/kubevirt/kubesecondarydns/pull/75)
> * [x]  [kubevirt/kubesecondarydns#73](https://github.com/kubevirt/kubesecondarydns/pull/73)
> * [x]  [k8snetworkplumbingwg/kubemacpool#441](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/441)
> * [x]  [kubevirt/cluster-network-addons-operator#1927](https://github.com/kubevirt/cluster-network-addons-operator/pull/1927)
> * [x]  [kubevirt/cluster-network-addons-operator#1928](https://github.com/kubevirt/cluster-network-addons-operator/pull/1928)
> * [x]  [kubevirt/bridge-marker#77](https://github.com/kubevirt/bridge-marker/pull/77)
> * [x]  [k8snetworkplumbingwg/ovs-cni#336](https://github.com/k8snetworkplumbingwg/ovs-cni/pull/336)
> 
> EDIT once we done, this is the list of what we should release / verify
> 
> * [ ]  kubeMacPool
> * [x]  kubeSecondaryDNS (released)
> * [x]  kubevirtIpamController
> * [ ]  linuxBridge
> * [ ]  macvtap
> * [x]  multus
> * [x]  multusDynamicNetworks (release was created)
> * [x]  OVS
> * [x]  Bridge marker
> * [ ]  ipam-ext
> * [ ]  CNAO

>
UPDATE:
We need to merge the following PRs to complete this effort::
https://github.com/kubevirt/ipam-extensions/pull/83
 https://github.com/kubevirt/macvtap-cni/pull/130
https://github.com/kubevirt/project-infra/pull/3855
https://github.com/kubevirt/cluster-network-addons-operator/pull/1950

**Comment by oshoval** [comment]
Hi, KMP was bumped on CNAO
Please check / tick the list above, once all are tested, lets create a new CNAO release
Thanks

**Comment by ashokpariya0** [comment]
> Hi, KMP was bumped on CNAO Please check / tick the list above, once all are tested, lets create a new CNAO release Thanks
>
>
Yes, it's all done. I've verified all the CNI plugins on s390x, so we’re good to proceed with the new CNAO release.

A couple of points to note for our reference:

1) We’ve pinned the Multus Dynamic version to v0.3.4 with [PR #1961](https://github.com/kubevirt/cluster-network-addons-operator/pull/1961), which does not support multi-architecture. The next version does support it, but we are sticking with v0.3.4 for now.
2) We don’t have multi-architecture support for pass-binding (but as we plan to remove it anyway, this isn’t a major concern ??).

Everything else looks good.

**Comment by RamLavi** [comment]
>We’ve pinned the Multus Dynamic version to v0.3.4 with https://github.com/kubevirt/cluster-network-addons-operator/pull/1961, which does not support multi-architecture. The next version does support it, but we are sticking with v0.3.4 for now.

@maiqueb do you plan to issue a new release for multus-dynamic?

**Comment by oshoval** [comment]
Thanks Ashok
About [2] we are good, it will be removed, and returned in other repo or so, and there it might be good to support multi arch.
But this @nirdothan and @EdDev will decide.
About [1] I think we don't need to be blocked imho unless Miguel plans soon to release a new release (assuming the blocker was fixed)

**Comment by ashokpariya0** [comment]
> Thanks Ashok About [2] we are good, it will be removed, and returned in other repo or so, and there it might be good to support multi arch. But this @nirdothan and @EdDev will decide. About [1] I think we don't need to be blocked imho unless Miguel plans soon to release a new release (assuming the blocker was fixed)
>
Okay, Thanks for the information. We can make a decision based on Miguel's response.

**Comment by oshoval** [comment]
multus dynamic 0.3.6 was merged

**Comment by oshoval** [comment]
Hi Ashok, thanks for the awesome effort,
please close this ticket once you good with it.

**Comment by ashokpariya0** [comment]
> Hi Ashok, thanks for the awesome effort, please close this ticket once you good with it.
>
Sure

**Comment by ashokpariya0** [comment]
Thank you everyone for all your help! A special thanks to @oshoval  for your outstanding support, initiative, and attention to detail. I truly appreciate your efforts in ensuring these were merged on time. Closing this issue as it’s now complete.
